chat-core/src/MongoDbEmbeddedContentStore.ts

```diff
@@ -0,0 +1,125 @@
+import { DatabaseConnection } from "./DatabaseConnection";
+import {
+  EmbeddedContent,
+  EmbeddedContentStore,
+  FindNearestNeighborsOptions,
+  WithScore,
+} from "./EmbeddedContent";
+import {
+  MakeMongoDbDatabaseConnectionParams,
+  makeMongoDbDatabaseConnection,
+  pageIdentity,
+} from "./MongoDbDatabaseConnection";
+import { strict as assert } from "assert";
+
+export function makeMongoDbEmbeddedContentStore({
+  connectionUri,
+  databaseName,
+}: MakeMongoDbDatabaseConnectionParams): EmbeddedContentStore &
+  DatabaseConnection {
```

Comment:
This might be useful for testing, so I'm not saying to remove it, but what does `DatabaseConnection` add here that `EmbeddedContentStore` doesn't have? Maybe `close()` (in which case - probably the Stores should have a `close()` function)

Reply:
i get what you're saying. will add it as an optional aspect b/c i can see not all stores needing a `.close()` method

Reply:
At first I wasn't sure but I think I like that!

Optional `cleanUp()` on every store?